src/main/java/org/folio/circulation/domain/policy/RemindersPolicy.java

```diff
@@ -0,0 +1,234 @@
+
+    private final int sequenceNumber;
+    private final Period period;
+    private final BigDecimal reminderFee;
+    private final String noticeFormat;
+    private final String noticeTemplateId;
+    private final String blockTemplateId;
+    private RemindersPolicy policy;
+
+    private ReminderConfig (int sequenceNumber, JsonObject entry) {
+      this.sequenceNumber = sequenceNumber;
+      this.period = Period.from(entry.getInteger(INTERVAL),
+        normalizeTimeUnit(entry.getString(TIME_UNIT_ID)));
+      this.reminderFee = getBigDecimalProperty(entry,REMINDER_FEE);
+      this.noticeFormat =  entry.getString(NOTICE_FORMAT);
+      this.noticeTemplateId= entry.getString(NOTICE_TEMPLATE_ID);
+      this.blockTemplateId = entry.getString(BLOCK_TEMPLATE_ID);
+    }
+
+    private ReminderConfig withPolicy(RemindersPolicy policy) {
+      this.policy = policy;
+      return this;
+    }
+
+    public NoticeFormat getNoticeFormat () {
+      return NoticeFormat.from(noticeFormat);
+    }
+
+    public boolean hasZeroFee () {
+      return reminderFee.doubleValue() == 0.0;
+    }
+
+    /**
+     * Calculates when the next reminder will become due, potentially avoiding closed days depending on the closed days setting.
+     * Takes a date in UTC, returns the result in UTC, and retains the time part of the offset date.
+     *
+     * @param offsetDate The loan due date/time, or the date/time of the most recent reminder, in UTC
+     * @param tenantTimeZone The zone ID for checking the dates in the right timezone
+     * @param servicePointId For retrieving the calendar with open/closed days
+     * @param calendars Access to stored calendars
+     * @return The resulting date/time in UTC.
+     */
+    public CompletableFuture<Result<ZonedDateTime>> nextNoticeDueOn(
+      ZonedDateTime offsetDate, ZoneId tenantTimeZone, String servicePointId, CalendarRepository calendars) {
+
+      log.debug("nextNoticeDueOn:: parameters offsetDate: {}, tenantTimeZone: {}" +
+          ", servicePointId: {}", () -> offsetDate, () -> tenantTimeZone, () -> servicePointId );
+
+      ZonedDateTime scheduledForDateTime = getPeriod().plusDate(offsetDate);
+      if (policy.canScheduleReminderUponClosedDay()) {
+        log.debug("nextNoticeDueOn: ignoring closed days");
+        return ofAsync(scheduledForDateTime);
+      } else {
+        log.debug("nextNoticeDueOn: taking closed days into account");
+        return getFirstComingOpenDay(scheduledForDateTime, tenantTimeZone, servicePointId, calendars);
+      }
+    }
+
+    private CompletableFuture<Result<ZonedDateTime>> getFirstComingOpenDay(
+      ZonedDateTime scheduledDate, ZoneId tenantTimeZone, String servicePointId, CalendarRepository calendars)  {
+
+      log.debug("getFirstComingOpenDay:: parameters scheduledDate: {}, tenantTimeZone: {}" +
+        ", servicePointId: {}", () -> scheduledDate, () -> tenantTimeZone, () -> servicePointId );
```

Comment:
All of these values (scheduledDate, tenantTimeZone, servicePointId) are objects with toString() methods, no need to use suppliers here. We're using suppliers to avoid unnecessary method calls. For example, something like `log.debug("{}", () -> formatTheObject(obj))` in this case `formatTheObject()` would be called regardless of the logging level. Even if it's set up to `info` and we're logging with `log.debug`, it would still be executed and this is what we're trying to avoid with suppliers. If we're just logging objects, the framework is smart enough to only call toString() when needed.

This comment actually applies to multiple logging lines in this PR, but I'm just going to leave one here to avoid clutter